### PR TITLE
[lag_interfaces] bug fixes and optimisation 

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

[lag_interface] bug fixes and optimisation 

##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

lag_interface

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by KB-perByte** [comment]
Module | statements | missing | excluded | branches | partial | coverage
-- | -- | -- | -- | -- | -- | --
Total | 208 | 4 | 0 | 44 | 3 | 97%
plugins/module_utils/network/ios/argspec/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/argspec/lag_interfaces/lag_interfaces.py | 4 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/config/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/config/lag_interfaces/lag_interfaces.py | 55 | 2 | 0 | 30 | 1 | 96%
plugins/module_utils/network/ios/facts/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/facts/lag_interfaces/lag_interfaces.py | 40 | 1 | 0 | 12 | 1 | 96%
plugins/module_utils/network/ios/rm_templates/lag_interfaces.py | 8 | 0 | 0 | 0 | 0 | 100%
plugins/modules/ios_lag_interfaces.py | 15 | 1 | 0 | 2 | 1 | 88%
tests/unit/modules/network/ios/test_ios_lag_interfaces.py | 86 | 0 | 0 | 0 | 0 | 100%

File: plugins/modules/ios_lag_interfaces.py [hunk 1]
```diff
@@ -192,14 +198,29 @@
 - name: Override device configuration of all interfaces with provided configuration
   cisco.ios.ios_lag_interfaces:
     config:
-    - name: 20
```
Comment (GomathiselviS) [inline plugins/modules/ios_lag_interfaces.py:195]: Will this break the existing playbooks?

File: tests/integration/targets/ios_lag_interfaces/vars/main.yaml [hunk 2]
```diff
@@ -1,16 +1,11 @@
 ---
 merged:
-  before:
-    - name: GigabitEthernet0/0
```
Comment (GomathiselviS) [inline tests/integration/targets/ios_lag_interfaces/vars/main.yaml:4]: For all the interface modules, we decided to populate all the interfaces present in the device ( though they do not have any configs under them) in the after and before keys. Can you please check this point with the team?

File: plugins/module_utils/network/ios/argspec/lag_interfaces/lag_interfaces.py [hunk 3]
```diff
@@ -1,69 +1,64 @@
...
                             "choices": [
                                 "auto",
-                                "on",
                                 "desirable",
+                                "on",
                                 "active",
                                 "passive",
                             ],
-                            "type": "str",
-                            "required": True,
```
[the remark this answers is not in the document]
Comment (KB-perByte) [reply plugins/module_utils/network/ios/argspec/lag_interfaces/lag_interfaces.py:56]: `mode` should not be a required key, that was perhaps a bug one can configure port channels without `mode` but with a `link`.

File: plugins/modules/ios_lag_interfaces.py (hunk 1 above)
Comment (KB-perByte) [reply plugins/modules/ios_lag_interfaces.py:195]: Nope, I have added a way to handle the older way of writing the playbook aka Port-Channel20 as 20. This is not right and we should not encourage hence taking it off the examples. But as it was introduced with the module previously it is kept likewise.

File: tests/integration/targets/ios_lag_interfaces/vars/main.yaml (hunk 2 above)
Comment (KB-perByte) [reply tests/integration/targets/ios_lag_interfaces/vars/main.yaml:4]: Yes, that is correct but specific to this module the facts would be misleading if we look at it that way,
The facts for `lag_interfaces` is a bit twisted i.e we have options Port-channels as the parent attribute under which all the other interfaces are termed as members, showing all the other interfaces in the top level with no data in members is basically wrong in terms of facts specific to lag_interfaces. This makes me think only blank Port-channels should be visible.

Comment (KB-perByte) [reply tests/integration/targets/ios_lag_interfaces/vars/main.yaml:4]: @GomathiselviS so now
for a configuration as 
```
interface Port-channel11
interface Port-channel22
interface Port-channel33
interface GigabitEthernet0/1
 shutdown
 channel-group 11 mode active
interface GigabitEthernet0/2
 shutdown
 channel-group 22 mode active
interface GigabitEthernet0/3
 shutdown
 channel-group 22 mode passive
interface GigabitEthernet0/4
 shutdown
 channel-group 22 mode active link 20
interface GigabitEthernet0/5
 shutdown
interface Port-channel44
```
We have 4 Port-channels and 5 GigEth and GigEth0/5 is not under a channel group 
we would produce the facts like -
```
"parsed": [
        {
            "members": [
                {
                    "member": "GigabitEthernet0/1",
                    "mode": "active"
                }
            ],
            "name": "Port-channel11"
        },
        {
            "members": [
                {
                    "member": "GigabitEthernet0/2",
                    "mode": "active"
                },
                {
                    "member": "GigabitEthernet0/3",
                    "mode": "passive"
                },
                {
                    "link": 20,
                    "member": "GigabitEthernet0/4",
                    "mode": "active"
                }
            ],
            "name": "Port-channel22"
        },
        {
            "members": [
                {}
            ],
            "name": "Port-channel33"
        },
        {
            "members": [
                {}
            ],
            "name": "Port-channel44"
        }
    ]
```
Not making a group for GigEth0/5 as it's not a valid group as per lag_interfaes and has no option to have _members_ within it.

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**